app/containers/MessageBox/index.js

```diff
@@ -601,13 +590,36 @@ class MessageBox extends Component {
 
 	showMessageBoxActions = () => {
 		ActionSheet.showActionSheetWithOptions({
-			options: this.messageBoxActions,
+			options: this.createListOfMessageBoxActions(),
 			cancelButtonIndex: FILE_CANCEL_INDEX
 		}, (actionIndex) => {
 			this.handleMessageBoxActions(actionIndex);
 		});
 	}
 
+	showFileActions = () => {
+		const { FileUpload_Enabled } = this.props;
+		if (FileUpload_Enabled) {
+			ActionSheet.showActionSheetWithOptions({
+				options: this.fileOptions,
+				cancelButtonIndex: FILE_CANCEL_INDEX
+			}, (actionIndex) => {
+				this.handleFileActionPress(actionIndex);
+			});
+		} else {
+			Alert.alert(I18n.t('Error_uploading'), I18n.t('error-message-file-upload-not-allowed'));
+		}
+	}
+
+	createListOfMessageBoxActions = () => {
+		const actions = [I18n.t('Cancel'), I18n.t('Create_Discussion')];
+		const { FileUpload_Enabled } = this.props;
+		if (FileUpload_Enabled) {
+			actions.push(I18n.t('Take_a_photo'), I18n.t('Take_a_video'), I18n.t('Choose_from_library'), I18n.t('Choose_file'));
+		}
```

Comment:
We're using this by [these](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/1780/files#diff-7fbe0fa0b4b111bf63127ce6e1366ffbR64-R69) actionIndexes, when `FileUpload` is enabled it's causing some confusion at the order of the actions, try to open this at the open server and click at `Choose File`.
Maybe you can do this more dynamically like we do at the `MessageActions`, this can be done at the constructor, since here we don't check anything with a promise. :D